src/app/utils/appendDimensionsToExtent.js

```diff
@@ -0,0 +1,18 @@
+export const appendDimensionsToExtent = (bib) => {
+  if (!bib.extent || bib.extent.length === 0) return
+  let extent = bib.extent[0]
+  let punctuationToAdd = ''
+  // Check if extent was cataloged with a semicolon already at the end:
+  const semicolon = (extent.slice(-2) === '; ' || extent.slice(-1) === ';')
+  if (semicolon) {
+    if (extent.slice(-1) !== ' ') punctuationToAdd += ' '
+  } else punctuationToAdd = '; '
+  if (bib.dimensions && bib.dimensions[0].length) {
+    // If there is a dimensions field, append  it to the extent and make sure they are separated by a semicolon and a space:
+    extent = extent + punctuationToAdd + bib.dimensions[0]
+  } else {
+    // If there is no dimensions field, remove the semicolon
+    extent = punctuationToAdd.length === 0 ? extent.slice(0, -2) : extent.slice(0, -1)
+  }
+  return [extent]
```

Comment:
I know this has already been reviewed, but it doesn't look like it handles cases where `dimensions` is missing:
```
> appendDimensionsToExtent({ extent: ['Extent'] })
[ 'Exten' ]
> appendDimensionsToExtent({ extent: ['Extent'], dimensions: [] })
Thrown:
TypeError: Cannot read property 'length' of undefined
```

I worry that the string hacking is difficult to follow and reason about. Seems like there are a lot of conditionals and substringing happening to accommodate whether or not there is already a semicolon join phrase in `extent`. I think this could be cleaner if we first removed that uncertainty and then joined, like this:
```
const appendDimensionsToExtent = (bib) => {
  if (!bib.extent || bib.extent.length === 0 || !bib.extent[0]) return;
  
  let parts = [ bib.extent[0].replace(/\s*;\s*$/, '') ]
  if (bib.dimensions && bib.dimensions[0]) parts.push(bib.dimensions[0])
  
  return parts.join('; ')
}
```